Fix bug in `check_empty_containers`

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63492**

Differential Revision: [D30402749](https://our.internmc.facebook.com/intern/diff/D30402749)

Commit:
Fix bug in `check_empty_containers`

[ghstack-poisoned]

Commit:
Update on "Fix bug in `check_empty_containers`"




[ghstack-poisoned]

ansley commented:
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

gmagogsfm reviewed (commented):
Good catch. This also reveals a hole in our test cases. Could we add a test case covering this bug to make sure the fix doesn't regress?

Commit:
Update on "Fix bug in `check_empty_containers`"


Differential Revision: [D30402749](https://our.internmc.facebook.com/intern/diff/D30402749)

[ghstack-poisoned]

ansley commented:
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

gmagogsfm commented:
Code looks good. However the test case comment isn't clear for future readers. Also there are some failures in CI, are they real or just flakes?

Commit:
Update on "Fix bug in `check_empty_containers`"


Differential Revision: [D30402749](https://our.internmc.facebook.com/intern/diff/D30402749)

[ghstack-poisoned]

ansley commented:
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

gmagogsfm commented on test/jit/test_isinstance.py:
```diff
@@ -310,3 +310,12 @@ def fn(x: Any):
+    def test_empty_container_special_cases(self):
+        # Should not throw "Boolean value of Tensor with no values is
+        # ambiguous" error
+        torch._jit_internal.check_empty_containers(torch.Tensor([]))
```
Thanks for adding the test case. I am curious how you came across this bug? Maybe we should that use case that leads you to this bug as a test case as well?

gmagogsfm replied:
@ansley ping

If that use case isn't suitable as unit test, I am OK with landing it as is too.
